src/documents/signals/handlers.py

```diff
@@ -866,6 +868,142 @@ def removal_action():
+            )
+        except Exception as e:
+            logger.exception(
+                f"Error occurred sending notification email: {e}",
+                extra={"group": logging_group},
+            )
+
+    def webhook_action():
+        title = (
+            document.title
+            if isinstance(document, Document)
+            else str(document.original_file)
+        )
+        doc_url = None
+        if isinstance(document, Document):
+            doc_url = f"{settings.PAPERLESS_URL}/documents/{document.pk}/"
+
+        try:
+            data = {}
+            if action.webhook_use_params:
+                try:
+                    for key, value in action.webhook_params.items():
+                        data[key] = parse_w_workflow_placeholders(
+                            value,
+                            document.correspondent.name
+                            if document.correspondent
+                            else "",
+                            document.document_type.name
+                            if document.document_type
+                            else "",
+                            document.owner.username if document.owner else "",
+                            timezone.localtime(document.added),
+                            document.original_filename or "",
+                            timezone.localtime(document.created),
+                            title,
+                            doc_url,
+                        )
+                except Exception as e:
+                    logger.error(
+                        f"Error occurred parsing webhook params: {e}",
+                        extra={"group": logging_group},
+                    )
+            else:
+                data = parse_w_workflow_placeholders(
+                    action.webhook_body,
+                    document.correspondent.name if document.correspondent else "",
+                    document.document_type.name if document.document_type else "",
+                    document.owner.username if document.owner else "",
+                    timezone.localtime(document.added),
+                    document.original_filename or "",
+                    timezone.localtime(document.created),
+                    title,
+                    doc_url,
+                )
+            headers = {}
+            if action.webhook_headers:
+                try:
+                    headers = {
+                        str(k): str(v) for k, v in action.webhook_headers.items()
+                    }
+                except Exception as e:
+                    logger.error(
+                        f"Error occurred parsing webhook headers: {e}",
+                        extra={"group": logging_group},
+                    )
+            if action.webhook_include_document:
+                with open(document.source_path, "rb") as f:
+                    files = {"file": (document.original_filename, f)}
```

Comment:
Since we should have it, I think we should set the content type, such as `files = {"file": (document.original_filename, f, document.mime_type)}`

The second example [here](https://www.python-httpx.org/quickstart/#sending-multipart-file-uploads)